### PR TITLE
docs: Add a basic how-to for switching to the new theme.

### PR DESCRIPTION
**Pull request by feanil**
With https://github.com/openedx/edx-sphinx-theme/issues/184 we deprecate
the edx-sphinx-theme so provide some docs for how people can migrate to
the new theme.

### CODE REVIEW

File: source/developers/how-tos/switch-to-the-sphinx-book-theme.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+   #. Remove your old theme(e.g. ``edx-sphinx-theme``) and add a line for the
+      sphinx-book-theme.
+
+      .. code::
+
+         - edx-sphinx-theme
+         + sphinx-book-theme
+
+   #. Run ``make upgrade``
```
Comment (nedbat) [inline source/developers/how-tos/switch-to-the-sphinx-book-theme.rst:40]: Is there a likelihood that something might go wrong here?

File: source/developers/how-tos/switch-to-the-sphinx-book-theme.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+#. Update conf.py
+
+   Add or update the following
+   .. code::
+
+      import os
+
+      html_theme_options = {
+
+          "repository_url": TODO: Add a URL Here,
```
Comment (nedbat) [inline source/developers/how-tos/switch-to-the-sphinx-book-theme.rst:57]:
```suggestion
          "repository_url": "https://github.com/openedx/the-name-of-your-repo",
```
This entry and the next could use some explanation

File: source/developers/how-tos/switch-to-the-sphinx-book-theme.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+          "repository_branch": TODO: Add the correct branch here,
+          "path_to_docs": "docs/",
+          "logo_only": True,
+          "home_page_in_toc": True,
+          "use_repository_button": True,
+          "use_issues_button": True,
+          "use_edit_page_button": True,
+          "extra_footer": """
```
Comment (nedbat) [inline source/developers/how-tos/switch-to-the-sphinx-book-theme.rst:65]: This one is complicated, and shouldn't be changed, correct?

File: source/developers/how-tos/switch-to-the-sphinx-book-theme.rst [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+              <a rel="license" href="https://creativecommons.org/licenses/by-sa/4.0/">
+                  <img
+                      alt="Creative Commons License"
+                      style="border-width:0"
+                      src="https://i.creativecommons.org/l/by-sa/4.0/80x15.png">
```
Comment (nedbat) [inline source/developers/how-tos/switch-to-the-sphinx-book-theme.rst:70]:
```suggestion
                      src="https://i.creativecommons.org/l/by-sa/4.0/80x15.png"/>
```

File: source/developers/how-tos/switch-to-the-sphinx-book-theme.rst [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+      # bug in the theme that needs to be fixed first.
+      # If you'd like you can temporarily copy the logo file to your `_static`
+      # directory.
+      html_logo = "https://logos.openedx.org/open-edx-logo-color.png"
+      html_favicon = "https://logos.openedx.org/open-edx-favicon.ico"
+
+      # Set the DJANGO_SETTINGS_MODULE if it's not set.
+      # Only if your project has a dependency on Django
+      if not os.environ.get('DJANGO_SETTINGS_MODULE'):
+         os.environ['DJANGO_SETTINGS_MODULE'] = 'test_utils.test_settings'
```
Comment (nedbat) [inline source/developers/how-tos/switch-to-the-sphinx-book-theme.rst:98]: This might also need some explanation.

**Review by nedbat — APPROVED** [review]
Mostly nits

File: source/developers/how-tos/switch-to-the-sphinx-book-theme.rst [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+   .. code::
+
+      import os
+
+      html_theme_options = {
+
+          "repository_url": TODO: Add a github URL Here, for example https://github.com/openedx/repo-name,
+          "repository_branch": TODO: Add the correct branch, for example 'main',
+          "path_to_docs": "docs/",
+          "logo_only": True,
```
[the remark this answers is not in the document]
Comment (feanil) [reply source/developers/how-tos/switch-to-the-sphinx-book-theme.rst:60]: ah, yea good call  I created https://github.com/openedx/docs.openedx.org/pull/345 to fix this.

**Pull request merged**